moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1924,16 +1924,27 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
 
   bool test_joint_space_jump = jump_threshold > 0.0;
 
+  Eigen::Quaterniond start_quaternion(start_pose.rotation());
+  Eigen::Quaterniond target_quaternion(rotated_target.rotation());
+  double distance = start_quaternion.dot(target_quaternion);
+  if (distance < 0)  // need to bring quaternions to same half sphere?
+  {
+    target_quaternion.x() *= -1.0;
+    target_quaternion.y() *= -1.0;
+    target_quaternion.z() *= -1.0;
+    target_quaternion.w() *= -1.0;
+  }
+
   // decide how many steps we will need for this trajectory
-  double distance = (rotated_target.translation() - start_pose.translation()).norm();
+  distance = std::max((rotated_target.translation() - start_pose.translation()).norm(),
```

Comment:
I really don't like that this is comparing distance in meters to distance radians. Wouldn't it be better to have an optional max_orientation_step parameter in addition to a max_cartesian_step?

Reply:
I agree. However, usually we want to stick with the existing API. In the intended rewrite of this whole API, we should consider two parameters...

Reply:
Perhaps we should add a `TODO` for this.

Reply:
I'm dealing with the same thing for #798  To deal with this I added additional function headers for the new api and used a struct to minimize code duplication

Reply:
I added the TODO  note.